### PR TITLE
Implemented the calendar labels.

### PR DESCRIPTION
**Pull request by bvitaliyg**
+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/master/CONTRIBUTING.md)

Remember to include the following changes:
+ [x] `UNRELEASED.md`
+ [x] `README.md`
+ [x] Tests
+ [x] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack/src/main/java/net/skyscanner/backpack/calendar/model/CalendarLabel.kt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.calendar.model
+
+data class CalendarLabel(
```
Comment (bvitaliyg) [inline Backpack/src/main/java/net/skyscanner/backpack/calendar/model/CalendarLabel.kt:21]: Public API

File: Backpack/src/main/java/net/skyscanner/backpack/calendar/view/BpkCalendarLabelsViewModel.kt [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+import android.graphics.Paint
+import android.text.TextPaint
+import android.text.TextUtils
+import androidx.core.content.ContextCompat
+import net.skyscanner.backpack.R
+import net.skyscanner.backpack.calendar.model.CalendarLabel
+import net.skyscanner.backpack.text.BpkText
+import org.threeten.bp.LocalDate
+
+internal class BpkCalendarLabelsViewModel(
```
Comment (bvitaliyg) [inline Backpack/src/main/java/net/skyscanner/backpack/calendar/view/BpkCalendarLabelsViewModel.kt:32]: The "renderer" of the labels. Manages and updates all the necessary information, preparing the data set to be drawn.

File: Backpack/src/main/java/net/skyscanner/backpack/calendar/view/MonthView.kt [hunk 3]
```diff
@@ -432,6 +439,14 @@ internal class MonthView @JvmOverloads constructor(
       y.toFloat(),
       monthNumberPaint
     )
+
+    labelsViewModel.draw(
```
Comment (bvitaliyg) [inline Backpack/src/main/java/net/skyscanner/backpack/calendar/view/MonthView.kt:443]: Will draw nothing if the labels not set, or will draw a dash if the labels are set, but there's no data for the current date.

File: app/src/androidTest/java/net/skyscanner/backpack/calendar/BpkCalendarTest.kt [hunk 4]
```diff
@@ -129,6 +131,33 @@ class BpkCalendarTest : BpkSnapshotTest() {
...
+      SelectionType.RANGE,
+      calendarLabels = mapOf(
+        LocalDate.of(initialStartDate.year, initialStartDate.month, initialStartDate.dayOfMonth + 1) to
+          CalendarLabel(text = "£10", style = CalendarLabel.Style.PriceHigh),
+        LocalDate.of(initialStartDate.year, initialStartDate.month, initialStartDate.dayOfMonth + 2) to
+          CalendarLabel(text = "£11", style = CalendarLabel.Style.PriceMedium),
+        LocalDate.of(initialStartDate.year, initialStartDate.month, initialStartDate.dayOfMonth + 3) to
+          CalendarLabel(text = "£12", style = CalendarLabel.Style.PriceLow),
+        LocalDate.of(initialStartDate.year, initialStartDate.month, initialStartDate.dayOfMonth + 4) to
+          CalendarLabel(text = "£900000000000000", style = CalendarLabel.Style.PriceLow),
```
Comment (shaundon) [inline app/src/androidTest/java/net/skyscanner/backpack/calendar/BpkCalendarTest.kt:153]: That's an expensive flight 😂

**Review by shaundon — COMMENTED** [review]
Code looks good (as much as I can understand it 😅 ), will just run locally now

**Review by shaundon — APPROVED** [review]
I've run it locally and it seems great. Happy for you to merge, but you might want to ask somebody who knows more about Android for a code review (maybe Haroldo?). Up to you though

**Comment by shaundon** [comment]
I just noticed CI is failing so we should fix that before merging too. I had a look at the logs but I'm not sure what's going wrong

**Review by haroldolivieri — APPROVED** [review]
🚀

**Comment by haroldolivieri** [comment]
Amazing work, thanks for this fast delivery :)

**Review by shaundon — APPROVED** [review]

**Pull request merged**